11-Backbonejs/Backbone-Views-of-Models-And-Events.md

```diff
@@ -1,103 +1,128 @@
+## Context for backbone-taskList
+// TODO: add more context about the task list.
+For this week, we'll be working with a codebase around making a task list.
+// TODO: add link
+We've provided some starter code. The starter code is located here.
+
+Pull the code (no need to fork) and open it.
 
-You may have noticed in the previous lectures, we have a jumble (technical term) of functions handling events, and drawing the tasklist to the browser etc.  This could quickly become a mess.  It's time for some structure!
+As always, run the following commands to get it going:
+```
+npm install
+npm start
+```
 
-We will first create two views, one for a single task item and a second for the collection of Tasks.  Each view will handle drawing and event handling for it's specific domain, and the view for the collection will utilize the smaller TaskView.  
+Take some time to look through the code. What code is already there? What displays on the browser?
 
-![Backbone Views](images/backbone-views.png)
+Our goal will be to take this code and make it take advantage of Backbone Views, which will also help us handle events.
 
-You can see a deployed version of this app [here:](http://cheezitman.github.io/tasklist-js)
+## What is a View?
+Backbone views are kind of middle-managers in the Backbone world, filling a similar role to controllers in Rails. A view's job is to coordinate between the data and the DOM. When a DOM event happens, it's the view's job to handle it and update the data as needed, and when the data changes it's the view's job to modify the DOM to match.
 
-## Creating A TaskView
+You may have noticed in the previous lectures, we have a jumble of functions handling events, and drawing to the browser etc. This could quickly become a mess. It's time for some structure!
 
-![Application Overview](images/Ada-TaskList-JS.png)
+For backbone-tasklist, we will create two views: one for a single task item and, later, a second view for the collection of Tasks.  Each view will handle drawing and event handling for it's specific domain, and the view for the collection will utilize the smaller view for the model.
 
-A TaskView will handle drawing an individual task item and responding to events that concern just that task.  To start we create a file in `app/views/task_view.js`  Each TaskView will only be concerned with rendering a specific task and dealing with events on that task.  We've indicated it in red in the image above.  
+![Backbone Views](images/backbone-views.png)
 
+## Creating a View for a Task (a TaskView)
+A TaskView will handle drawing an individual task item and responding to events that concern just that task.  To start, we create a file in `app/views/task_view.js`  Each TaskView will only be concerned with rendering a specific task and dealing with events on that task.
 
-```JavaScript
+```javascript
 import Backbone from 'backbone';
 import _ from 'underscore';
 import $ from 'jquery';
 import Task from '../models/task.js';
```

Comment:
That will work, but it will also draw complaints from the linter, I haven't figured out how to tell the linter to use the src folder as a library (which is what it assumes if it doesn't have the `.`).

Reply:
Oh hmm, that's a good point. I'm not sure what we would need to change about `.eslintrc`, but somehow I guess we should be telling it to use the local `.babelrc` from our project, if it's not.